docs/contributing/coding_guide.rst

```diff
@@ -880,34 +880,54 @@ outputs in the notebook* but store it in the repository with a
 :file:`preexecuted_` prefix (e.g.,
 :file:`preexecuted_full_3d_mhd_chaotic_turbulence_simulation.ipynb`).
 
-Compatibility with Prior Versions of Python, NumPy, and Astropy
-===============================================================
+Python and dependency version support
+=====================================
 
 PlasmaPy releases will generally abide by the following standards, which
-are adapted from `NEP 29`_ for the support of old versions of |Python|,
-|NumPy|, and |Astropy|.
+are adapted from `NEP 29`_ and `SPEC 0`_ for the support of old versions
+of |Python|, |NumPy|, |Astropy|, and other dependencies.
 
-* PlasmaPy should support at least the minor versions of Python
-  initially released 42 months prior to a planned project release date.
+* PlasmaPy should support the minor versions of Python initially
+  released 42 months prior to a release. Because :pep:`602` establishes
+  that Python releases will generally occur in October of each year,
+  the minimum required version fo Python should be increased for the
+  first release of PlasmaPy after April each year.
 
-* PlasmaPy should support at least the 3 latest minor versions of
-  Python.
+* PlasmaPy should support minor versions of its dependencies that were
+  released in the 24 months prior to each release of PlasmaPy.
+  Exceptions to this guideline should only be made when there are good
+  reasons to do so, such as impactful bugfixes or major improvements to
+  upstream functionality.
+
+  .. caution::
 
-* PlasmaPy should support minor versions of NumPy initially released
-  in the 24 months prior to a planned project release date or the
-  oldest version that supports the minimum Python version (whichever is
-  higher).
+     Even if a dependency is unlikely to be shared with packages
+     installed alongside PlasmaPy, that dependency can have strict
+     requirements that do cause conflicts. For example, requiring the
+     newest version of voila_ once caused dependency conflicts with
+     other packages in the heliopythoniverse because voila_ had strict
+     dependencies on packages in the Jupyter ecosystem.
 
 * PlasmaPy should support at least the 3 latest minor versions of
-  NumPy and Astropy.
+  |NumPy| and |Astropy|.
 
-The required major and minor version numbers of upstream packages may
-only change during major or minor releases of PlasmaPy, and never during
-patch releases.
+* The required major and minor version numbers of upstream packages may
+  only change during major or minor releases of PlasmaPy, and not during
+  patch releases.
+
+* The ``tests`` and ``docs`` dependency sets are required for running
+  tests and building documentation, but are not required for package
+  installation. Consequently, it is not necessary to support older
+  versions of packages that are only in these optional dependency sets.
```

Comment:
In addition, our tests are run and our docs are built in virtual environments created by tox or nox specifically for a single purpose.